inference_segmentation.py

```diff
@@ -112,15 +112,14 @@ def auto_batch_size_finder(datamodule, device, model, single_class_mode=False, m
     src_size = datamodule.inference_dataset.src.height * datamodule.inference_dataset.src.width
     bs_min, bs_max, bs_step = 4, 256, 4
     window_spline_2d = create_spline_window(datamodule.patch_size).to(device)
-    _tqdm = tqdm(range(bs_min, bs_max, bs_step), desc=f"Finding batch size filling GPU to {max_used_ram} % or less")
-    for trial in _tqdm:
+    for trial in tqdm(range(bs_min, bs_max, bs_step), desc=f"Finding batch size filling GPU to {max_used_ram} % or less"):
         batch_size_trial = trial
-        _tqdm.set_postfix_str(f"Trying: {batch_size_trial}")
         datamodule.batch_size = batch_size_trial
         if batch_size_trial*datamodule.patch_size**2 > src_size:
             logging.info(f"Reached maximum batch size for image size. "
                          f"Batch size tuned to {batch_size_trial-bs_step}.")
-            return batch_size_trial-bs_step
+            datamodule.batch_size = batch_size_trial-bs_step
+            return window_spline_2d
```

Comment:
why return the window spline instead of new batch size? The window spline is affected by patch_size, but I don't see the relation to batch size.

Reply:
Suggested solution:
- define window spline prior to auto_batch_size_finder
- pass window to auto_batch_size_finder as parameter
- auto_batch_size_finder doesn't return anything since the datamodule.batch_size is directly modified by this function
```
    window_spline_2d = create_spline_window(chip_size).to(device)

    if auto_batch_size and device.type != "cpu":
        auto_batch_size_finder(datamodule=dm,
                                            device=device,
                                            model=model,
                                            single_class_mode=single_class_mode,
                                            max_used_ram=auto_bs_threshold,
                                            window_spline_2d=window_spline_2d
                                            )
```